test/conformance/exp_command_buffer/update/local_memory_update.cpp

```diff
@@ -1105,3 +1105,223 @@ TEST_P(LocalMemoryMultiUpdateTest, UpdateWithoutBlocking) {
+            kernel, index_order[current_index++], nullptr, shared_ptrs[0]));
+
+        // Index 4 is X
+        ASSERT_SUCCESS(urKernelSetArgPointer(
+            kernel, index_order[current_index++], nullptr, shared_ptrs[1]));
+        // Index 5 is Y
+        ASSERT_SUCCESS(urKernelSetArgPointer(
+            kernel, index_order[current_index++], nullptr, shared_ptrs[2]));
+
+        // Index 1 is local_mem_b arg
+        ASSERT_SUCCESS(urKernelSetArgLocal(kernel, index_order[current_index++],
+                                           local_mem_b_size, nullptr));
+        if (backend == UR_PLATFORM_BACKEND_HIP) {
+            ASSERT_SUCCESS(urKernelSetArgValue(
+                kernel, index_order[current_index++], sizeof(hip_local_offset),
+                nullptr, &hip_local_offset));
+            ASSERT_SUCCESS(urKernelSetArgValue(
+                kernel, index_order[current_index++], sizeof(hip_local_offset),
+                nullptr, &hip_local_offset));
+            ASSERT_SUCCESS(urKernelSetArgValue(
+                kernel, index_order[current_index++], sizeof(hip_local_offset),
+                nullptr, &hip_local_offset));
+        }
+
+        // Index 0 is local_mem_a arg
+        ASSERT_SUCCESS(urKernelSetArgLocal(kernel, index_order[current_index++],
+                                           local_mem_a_size, nullptr));
+
+        // Hip has extra args for local mem at index 1-3
+        if (backend == UR_PLATFORM_BACKEND_HIP) {
+            ASSERT_SUCCESS(urKernelSetArgValue(
+                kernel, index_order[current_index++], sizeof(hip_local_offset),
+                nullptr, &hip_local_offset));
+            ASSERT_SUCCESS(urKernelSetArgValue(
+                kernel, index_order[current_index++], sizeof(hip_local_offset),
+                nullptr, &hip_local_offset));
+            ASSERT_SUCCESS(urKernelSetArgValue(
+                kernel, index_order[current_index++], sizeof(hip_local_offset),
+                nullptr, &hip_local_offset));
+        }
+    }
+};
+
+struct LocalMemoryUpdateTestOutOfOrder : LocalMemoryUpdateTestBaseOutOfOrder {
+    void SetUp() override {
+        UUR_RETURN_ON_FATAL_FAILURE(
+            LocalMemoryUpdateTestBaseOutOfOrder::SetUp());
+
+        // Append kernel command to command-buffer and close command-buffer
+        ASSERT_SUCCESS(urCommandBufferAppendKernelLaunchExp(
+            updatable_cmd_buf_handle, kernel, n_dimensions, &global_offset,
+            &global_size, &local_size, 0, nullptr, 0, nullptr, 0, nullptr,
+            nullptr, nullptr, &command_handle));
+        ASSERT_NE(command_handle, nullptr);
+
+        ASSERT_SUCCESS(urCommandBufferFinalizeExp(updatable_cmd_buf_handle));
+    }
+
+    void TearDown() override {
+        if (command_handle) {
+            EXPECT_SUCCESS(urCommandBufferReleaseCommandExp(command_handle));
+        }
+
+        UUR_RETURN_ON_FATAL_FAILURE(
+            LocalMemoryUpdateTestBaseOutOfOrder::TearDown());
+    }
```

Comment:
Nothing to do right now, but heads up this will conflict with https://github.com/oneapi-src/unified-runtime/pull/2578